atc/testhelpers/walker.go

```diff
@@ -0,0 +1,168 @@
+package testhelpers
+
+import "github.com/concourse/concourse/atc"
+
+type PlanWalker func(atc.Plan) (atc.Plan, error)
+
+// Walks the plan tree structure calling walker for each encountered Plan,
+// including the root.
+func WalkPlan(plan atc.Plan, walker PlanWalker) (atc.Plan, error) {
```

Comment:
This looks a lot like `atc.Plan.Each`:

https://github.com/concourse/concourse/blob/0565fd9ebe7cbd86b1f85f6c49d0ea52e145b421/atc/plan.go#L36

Maybe just worth modifying `atc.Plan.Each` to take a `func(*atc.Plan) error` instead?

Reply:
Good call. I don't strictly need to return an error in the new matcher that I introduced. I will modify both matchers to use `plan.Each`.

Reply:
I pushed the changes. Please, take another look.